pyxem/data/_data.py

```diff
@@ -0,0 +1,311 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2023 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+import pooch
+from pyxem.data._registry import _file_names_hash, _urls
+import hyperspy.api as hs
+from pathlib import Path
+from pyxem.release_info import version
+from typing import Optional, Union
+import os
+
+# Create a downloader object
+marshall = pooch.create(
+    path=pooch.os_cache("pyxem"),
+    base_url="",
+    version=version.replace(".dev", "+"),
+    version_dev="develop",
+    env="PYXEM_DATA_DIR",
```

Comment:
I suggest to describe in the docs how the cache location can be set using this environment variable. We've done so in [kikuchipy's docs](https://kikuchipy.org/en/stable/reference/generated/kikuchipy.data.html).